src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -245,3 +245,26 @@ def test_reset_with_default(session):
     assert non_default_arb_sample_rate == 100000000.0
     session.reset_with_defaults()
     assert session.arb_sample_rate == 250000000.0
+
+
+def test_write_binary_waveform(session):
+    session.write_binary16_waveform(session.allocate_waveform(10), [0, 0, 0, 1, 1, 1, 2, 2])
+
+
+'''
+(TODO) Jaleel , check it after issue #538 fixed
+def test_set_waveform_next_write_position(session):
+    session.set_waveform_next_write_position(session.allocate_waveform(10), nifgen.RelativeTo.START, 5)  # Enable after RelativeTo enum added to enums_addon.py
+
+
+def test_export_signal(session):
+    expected_trigger_terminal = "PXI_Trig0"
+    session.export_signal(nifgen.ExportSignal.START_TRIGGER, "", expected_trigger_terminal)  # Enable after issue #538 fixed
+    assert expected_trigger_terminal == session.exported_start_trigger_output_terminal
+
+
+def test_write_waveform_from_filei64(session):
+    session.arb_sample_rate = 40000000
+    session.create_waveform_from_file_i16(os.path.join(os.getcwd(), 'systemtest_dependencies', 'SineI16BigEndian_1000.bin'), nifgen.ByteOrder.BIG_ENDIAN)  # Enable after issue #538 fixed
```

Comment:
I would directly put the file next to the system test script, I don't think we need a separate folder.
Alternatively, there may be a really compact way to create this file from the the test when we need it, you could put it in the temp directory.